### ISSUE TITLE
Is there a plan of migration to material-ui v5?

### ISSUE THREAD

**Issue by yaminyaylo** [gregnb/mui-datatables#1513]
The issue is when I updated the Material-UI library to the latest version I found out that MUI-datatables doesn't support the Material-UI library v5.

Updated Material-UI libs to the following versions:
*  "@material-ui/core": "5.0.0-alpha.9",
*  "@material-ui/icons": "5.0.0-alpha.7",
*  "@material-ui/lab": "5.0.0-alpha.9",
*  "@material-ui/pickers": "^4.0.0-alpha.12",
*  "@material-ui/styles": "5.0.0-alpha.8",

## Expected Behavior
No errors and crashed on the Table filter action.

## Current Behavior

There are errors and crashed on the Table filter action press.

## Steps to Reproduce (for bugs)

1. update material-ui and it's dependant libs to the latest
2. fix all breaking changes in your app
3. go to the page you have a table based on MUI-datatables
4. suppose it has passed data inside the table and **filter** option is true
5. click on the Filter icon button
[](url)

## Your Environment

| Tech         | Version |
|--------------|---------|
| Material-UI  |    5.0.0-alpha.9     |
| MUI-datatables  |   3.4.1      |
| React        |    16.13.1     |
| browser      |     Chrome (latest)    |
| etc          |         |

**Comment by patorjk** [comment]
I would accept pull requests to get the table working on version 5. If it requires breaking changes, it'd require a version bump and would probably be worth waiting until the official version 5 is released. I have not yet used version 5.

Also worth mentioning: The Material UI team is actually going to release an official data-table component for version 5.0. I talked with the lead Material UI dev a few months ago about it, and chatted back and forth through email around 6-7 weeks ago with the developer they have working on the official data-table (I tried the table out and then made a bunch of suggestions - the table looks pretty good and the developer seemed very receptive to my feedback). The goal is to get rid of the need for 3rd party data-tables like mui-datatables and material-table. However, the unfortunate thing is that many of the features that mui-datatables currently supports will be locked behind a paid enterprise version (at least that was the case 6-7 weeks ago), so I'm assuming there will still a place for mui-datatables and material-table, but I've sort of taken a wait-and-see approach by not doing too much development, as I'm not sure what direction the community will go. However, I'm open to PRs from other devs.

**Comment by yaminyaylo** [comment]
@patorjk thanks a lot for a very quick response and explanation.

I think we as a team will think and negotiate about what to do.

**Comment by patorjk** [comment]
@wdh2100 has taken over as the main maintainer. I wouldn't mind helping out though if he needs some assistance with this.

**Comment by davidcraddock** [comment]
so will this project be updated to v5?

**Comment by wdh2100** [comment]
@davidcraddock 

Yes, I hava plan to proceed.
When material-ui enter the stable phase, I will switch to v5.

https://github.com/gregnb/mui-datatables/issues?q=is%3Aissue+label%3A%22material+ui+5.0%22+

**Comment by davidcraddock** [comment]
@wdh2100 thanks for the clarification

**Comment by yaminyaylo** [comment]
@wdh2100 that's great news, looking forward to material-UI v.5 entering the stable phase

**Comment by artivilla** [comment]
Given [material-ui release v5.0.0-beta.0](https://github.com/mui-org/material-ui/releases) now encourages new projects to start with v5 I think it would be beneficial to support v5. Thoughts?

**Comment by lapico** [comment]
We really need that, it does not make sense to start a project with v4 anymore...

**Comment by lapico** [comment]
After days of struggling with Material v5 Datagrid component, I rolled back to Material v4 & mui-datatables.

Comparing the two libraries really showed how mui-datatables is relevant, it's just so good !

**Comment by yaminyaylo** [comment]
yeah, all our codebase is waiting to migrate to material-ui v5 beta (actually we now using "5.0.0-alpha.5"), but mui-datatables is broken after increasing the version of the material-ui core higher than  "5.0.0-alpha.5" :(

**Comment by davidcraddock** [comment]
Yeah I had to update my forked repository above to use the latest beta.2 material-ui today until there’s an official release

**Comment by MendelBak** [comment]
@yaminyaylo Is this library still maintained? I noticed there hasn't been many PRs or issues closed in the past few months. Is this because you're waiting till v5 of MUI comes out, and to migrate mui-datables to that, before allowing changes?

**Comment by yaminyaylo** [comment]
> @yaminyaylo Is this library still maintained? I noticed there hasn't been many PRs or issues closed in the past few months. Is this because you're waiting till v5 of MUI comes out, and to migrate mui-datables to that, before allowing changes?

I hope it still maintained because it will be a sad story :) 
I'm waiting that `mui-datatables` devs will bump up `mui-datatables` to material-ui v5.

**Comment by wdh2100** [comment]
> @wdh2100 thanks for the clarification

@davidcraddock  @lapico @yaminyaylo @MendelBak

I'll try to make an beta version

The target version is `@material-ui/core@5.0.0-beta.4`

>Given material-ui release v5.0.0-beta.0 now encourages new projects to start with v5 I think it would be beneficial to support v5. Thoughts?

@artivilla 

thanks!

**Comment by luluhoc** [comment]
Yeah I updated my app to v5 material just to find issue with mui-datatables :( please update it.

**Comment by gnowland** [comment]
@wdh2100 looks like [v5 stable is two weeks away](https://github.com/mui-org/material-ui/issues/27704#issuecomment-900074737), will this be ready?

**Comment by zxhmike** [comment]
Hi @wdh2100 ,

Our team spent some time upgrading mui-datatables to Material-UI 5. The preliminary PR can be found here: https://github.com/gregnb/mui-datatables/pull/1772

Sorry, I just saw the updates on this issue that you were making some effort to do a similar upgrade. Hopefully, we did not step on your toes.

Please review the PR and give us feedback. We were able to bring up the examples page, but there are some known issues that we are looking into, which can be found in the description of the PR.

Thank you! Looking forward to your reply.

**Comment by 293nav** [comment]
Any update on upgrading this library to support mui v5. Material-ui has released a new stable version. I noticed that in the stable version the imports are now made from @mui/material instead of @material-ui/

**Comment by 3BDO-debug** [comment]
I got it working with material ui v5 successfully by fixing some dependencies and little tweaks check it out now.

https://github.com/3BDO-debug/MUIDataTable-MUI-V5

**Comment by yaminyaylo** [comment]
Huge thanks to the team who migrated to material UI v5! You ROCK!!!

Can someone close the issue? 

CC @patorjk @wdh2100
